python/private/bzlmod/pip.bzl

```diff
@@ -174,6 +178,34 @@ def _create_whl_repos(module_ctx, pip_attr, whl_map, whl_overrides):
         whl_group_mapping = {}
         requirement_cycles = {}
 
+    index_urls = {}
+    host_cpu, host_os = None, None
+    if pip_attr.experimental_index_url:
+        if pip_attr.download_only:
+            fail("Currently unsupported to use `download_only` and `experimental_index_url`")
+
+        index_urls = simpleapi_download(
+            module_ctx,
+            index_url = pip_attr.experimental_index_url,
+            # TODO @aignas 2024-03-28: support index overrides for specific packages
+            # We should never attempt to join index contents ourselves.
+            index_url_overrides = pip_attr.experimental_index_url_overrides or {},
+            sources = [requirements_lock_content],
+            envsubst = pip_attr.envsubst,
+            cache = simpleapi_cache,
+        )
+
+        for constraint in HOST_CONSTRAINTS:
+            if "@platforms//cpu:" in constraint:
+                _, _, host_cpu = constraint.partition(":")
+            elif "@platforms//os:" in constraint:
+                _, _, host_os = constraint.partition(":")
+
+        if not (host_os and host_cpu):
+            fail("Don't have host information")
+
+    major_minor = _major_minor_version(pip_attr.python_version)
+
     # Create a new wheel library for each of the different whls
     for whl_name, requirement_line in requirements:
```

Comment:
I'll think how to refactor that. I think it may be probably better to wait until we are fetching wheels for all platforms as that will cause some churn of the code as well.

Reply:
Fair, SGTM